I've also tested this PR and it works like a charm with `vcpkg install opencv[cuda,qt,contrib,eigen,jpeg,png]`.

Please also note that without this PR the port may not compile.  
In fact, the [recent preference of Ninja build system](https://github.com/Microsoft/vcpkg/commit/471f7dc9700f7618aa16274fb9961e4e068f7656#diff-4fa98fbb3179d5f3752868d923d9c9aaR207), which uses full throttle CPU with parallel compilation, may trigger the following error:
```
C:/Users/<user>/AppData/Local/Temp/tmpxft_0001f7cc_00000000-9_minmaxloc.compute_70.cudafe1.cpp: fatal error C1041: cannot open program database 'C:\Users\<user>\vcpkg\buildtrees\opencv\x64-windows-rel\modules\cudaarithm\CMakeFiles\cuda_compile.dir\src\cuda\vc140.pdb'; if multiple CL.EXE write to the same .PDB file, please use /FS
CMake Error at cuda_compile_generated_minmaxloc.cu.obj.cmake:264 (message):
  Error generating file
  C:/Users/<user>/vcpkg/buildtrees/opencv/x64-windows-rel/modules/cudaarithm/CMakeFiles/cuda_compile.dir/src/cuda/./cuda_compile_generated_minmaxloc.cu.obj
```
As pointed out by the compiler, the problem seems to be related to a missing `/FS` ([MSDN](https://msdn.microsoft.com/en-us/library/dn502518.aspx)) flags that avoids concurrent IOs on PDB files.  
Since the current portfile points to `OpenCV 3.4.0` we have that [`/FS` extra compiler option](https://github.com/opencv/opencv/blob/3.4.0/cmake/OpenCVCompilerOptions.cmake#L249) is not passed to the compiler, while `OpenCV 3.4.1` [adds it corrcetly](https://github.com/opencv/opencv/blob/3.4.1/cmake/OpenCVCompilerOptions.cmake#L224).

Thanks @cenit for the effort and the PR. Hope to see it merged soon!

--- a/ports/opencv/CONTROL
+++ b/ports/opencv/CONTROL
@@ -24,6 +24,7 @@ Build-Depends: eigen3, glog, gflags, ceres
 Description: opencv_sfm module
 
 Feature: contrib
+Build-Depends: opencv[flann]
 Description: opencv_contrib module
 
 Feature: cuda

--- a/ports/opencv/portfile.cmake
+++ b/ports/opencv/portfile.cmake
@@ -66,6 +66,12 @@ if("contrib" IN_LIST FEATURES)
     SHA512 c16e60a6c4bb4de3ab39b876ae3c3f320ea56f69c93e9303bd2dff8760841dcd71be4161fff8bc71e8fe4fe8747fa8465d49d6bd8f5ebcdaea161f4bc2da7c93
   )
 
+  vcpkg_download_distfile(TINYDNN_ARCHIVE
+    URLS "https://github.com/tiny-dnn/tiny-dnn/archive/v1.0.0a3.tar.gz"
+    FILENAME "opencv-cache/tiny_dnn/adb1c512e09ca2c7a6faef36f9c53e59-v1.0.0a3.tar.gz"
+    SHA512 5f2c1a161771efa67e85b1fea395953b7744e29f61187ac5a6c54c912fb195b3aef9a5827135c3668bd0eeea5ae04a33cc433e1f6683e2b7955010a2632d168b
+  )
+
   function(download_opencv_3rdparty ID COMMIT HASH)
     if(NOT EXISTS "${DOWNLOADS}/opencv-cache/${ID}/${COMMIT}.stamp")
       vcpkg_download_distfile(OCV_DOWNLOAD

